### PR TITLE
Remove cosmos/keyring

### PR DESCRIPTION
**Pull request by rootulp**
Resolves https://github.com/celestiaorg/celestia-app/issues/538

cosmos/keyring hasn't pulled upstream changes from 99designs/keyring yet and the latest release of 99designs/keyring (1.2.1) appears to contain a fix for MacOS warnings: https://github.com/99designs/keyring/issues/107

Links
- https://github.com/cosmos/keyring/tags
- https://github.com/99designs/keyring/releases/tag/v1.2.1

`make test` passes after this change and it looks like other Cosmos chains don't need this replace:
- https://github.com/osmosis-labs/osmosis/blob/main/go.mod#L274-L285=
- https://github.com/CosmosContracts/juno/blob/main/go.mod#L178-L182=

Does anyone know if it's still necessary? I couldn't find the justification in https://github.com/celestiaorg/celestia-app/pull/127 which added it.

### CODE REVIEW

File: go.sum
Comment (rootulp) [reply go.sum:80]: It only makes sense to do in celestiaorg/cosmos-sdk if https://github.com/cosmos/cosmos-sdk/pull/12576 merges so will keep this in draft PR for now

**Comment by adlerjohn** [comment]
cc @evan-forbes

**Comment by evan-forbes** [comment]
I don't have a mac to test, but this is plenty good enough for me https://github.com/cosmos/cosmos-sdk/pull/12576

we merge this so that we don't have to wait for the next cosmos-sdk upgrade I think

**Comment by rootulp** [comment]
https://github.com/cosmos/cosmos-sdk/pull/12576 merged but I don't think this PR is ready b/c it won't resolve warnings on macOS until we perform a similar `replace` removal in [celestiaorg/cosmos-sdk](https://github.com/celestiaorg/cosmos-sdk/blob/release/v0.46.x-celestia/go.mod#L158=).

I'd rather not add a direct dependency in celestia-app on keyring v1.2.1 b/c it isn't a true direct dependency of this repo.

Defer to @evan-forbes on our strategy for upgrading celestiaorg/cosmos-sdk. Options I can see:
1. Wait until a cosmos-sdk release is cut and then pull upstream changes to celestiaorg/cosmos-sdk.
2. If it isn't a maintenance burden, I can cherry-pick the commit to celestiaorg/cosmos-sdk but I assume we want to minimize the diff between celestiaorg/cosmos-sdk and cosmos/cosmos-sdk.

**Comment by rootulp** [comment]
Blocked on a release of https://github.com/celestiaorg/cosmos-sdk

**Comment by rootulp** [comment]
Based on https://github.com/cosmos/cosmos-sdk/pull/12707 it looks like it will be included in v0.46.0 so agreed, thanks @evan-forbes !

**Comment by rootulp** [comment]
On second thought, I think we still want to remove the `replace` in this repo

**Comment by rootulp** [comment]
Status: still blocked on a release of https://github.com/celestiaorg/cosmos-sdk

**Comment by rootulp** [comment]
Release was cut: https://github.com/celestiaorg/cosmos-sdk/releases/tag/v1.2.0-sdk-v0.46.0

Will update this PR after https://github.com/celestiaorg/celestia-app/pull/472 merges

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**
